Improve routing/templates

michaelrkn opened the pull request:
Improved explanation of default behavior. Added more examples. Used headings to split up different parts.

locks commented on source/routing/rendering-a-template.md:
````diff
@@ -1,21 +1,19 @@
-  this.route('post');
-});
-```
+Each template will be rendered into the `{{outlet}}` of its parent route's
````
this is tricky because you can name the route `posts.new` without it being the child, kinda. would it benefit from a small router example?

mixonic replied:
:+1:

locks commented on source/routing/rendering-a-template.md:
````diff
@@ -25,16 +23,7 @@ export default Ember.Route.extend({
-  }
-});
-```
+## Named Outlets
````
We should confirm if components replace all use cases of named outlets and if this section should be removed from the guide. cc @rwjblue

mixonic replied:
@ef4 is also bullish on that story.

ef4 replied:
I am fairly convinced that we don't want to encourage named outlets for anything. As an alternative, we need to write examples like a basic sidebar and modal using component + service.

mixonic commented on source/routing/rendering-a-template.md:
```diff
@@ -1,21 +1,19 @@
-One of the most important jobs of a route handler is rendering the
+One job of a route handler is rendering the
```
:clap:

trek replied:
:clap: :clap:

michaelrkn commented:
Thanks for the feedback! I've updated the PR. Re "we need to write examples like a basic sidebar and modal using component + service", it sounds like that's not appropriate for the Routing section, so I've just removed the stuff on named outlets without replacement. Maybe somebody with a better understanding of how the above example might work can open an issue to write a guide for it?
